packages/svelte/src/compiler/phases/3-transform/server/visitors/VariableDeclaration.js

```diff
@@ -25,8 +25,14 @@ export function VariableDeclaration(node, context) {
 			}
 
 			if (rune === '$props') {
+				let has_rest = false;
 				// remove $bindable() from props declaration
 				const id = walk(declarator.id, null, {
+					RestElement(node, context) {
+						if (context.path.at(-1) === declarator.id) {
+							has_rest = true;
+						}
+					},
```

Comment:
We should also cover this case
```svelte
<script>
	let rest = $props();
</script>
{@render rest.children?.()}
<ul>
	{#each Object.getOwnPropertyNames(rest) as n}
			<li>{n}</li>
	{/each}
</ul>
```
that is still [broken in this PR](https://svelte-5-preview-5j4omcg69-svelte.vercel.app/#H4sIAAAAAAAAEz2MwarCMBBFfyXMc9FCafc19ukPqHvroi8dNZJOw2SqSMi_SxQed3Uv554IF-swQHuKQMOE0MLOe6hAXj6X8EAnCBWEeWGTFx0MWy9dT704FMUYRG3UyvPsQ1Gue9LNPxK3jDQif6ja3KwbGem3LsrUk17cxxJ_cDA3dfi7o5H6inJ40pFnjyyv_TBhKPK7VENQlPIhRzvbRUq6cfYrabIkW5vFdVDBNI_2YnGEVnjBdE5v2YEb2OsAAAA=)

Reply:
Pushed a fix, thanks for flagging!